### PR TITLE
Add --allow-unsecure-http-clients and --instance-name params...

### PR DESCRIPTION
**Pull request by 1st1**
... to the EdgeDB server.

Please see the individual commits.

### CODE REVIEW

File: edb/server/args.py [hunk 1]
```diff
@@ -95,7 +95,9 @@ class ServerConfig(NamedTuple):
     tls_cert_file: Optional[pathlib.Path]
     tls_key_file: Optional[pathlib.Path]
     generate_self_signed_cert: bool
-    allow_cleartext_connections: bool
+
+    allow_unsecure_binary_clients: bool
```
Comment (elprans) [inline edb/server/args.py:99]: insecure?

File: edb/server/protocol/protocol.pyx [hunk 2]
```diff
@@ -133,8 +136,9 @@ cdef class HttpProtocol:
                 return
             else:
                 # HTTP.
+                print('!!!', self.allow_unsecure_http_clients)
```
Comment (elprans) [inline edb/server/protocol/protocol.pyx:139]: Forgotten debug print

Comment (1st1) [reply edb/server/protocol/protocol.pyx]: fixed

File: edb/server/args.py (hunk 1 above)
Comment (1st1) [reply edb/server/args.py]: yeah, fixed

**Review by elprans — APPROVED** [review]

File: tests/test_server_ops.py [hunk 3]
```diff
@@ -403,6 +404,12 @@ async def test_server_ops_no_cleartext(self):
                 self.assertIn('location', resp_headers)
                 self.assertTrue(
                     resp_headers['location'].startswith('https://'))
+
+                self.assertIn('strict-transport-security', resp_headers)
+                # By default we enforce HTTPS via HSTS on all routes.
+                self.assertEqual(
+                    resp_headers['strict-transport-security'],
+                    'max-age=31536000; includesubdomains; preload')
```
Comment (elprans) [inline tests/test_server_ops.py:412]: I don't think we should force `includeSubdomains` and `preload` here. We have no business in enforcing HSTS opinions on the entire domain on which EdgeDB is running.

Comment (1st1) [reply tests/test_server_ops.py:412]: good catch, makes sense

Comment (1st1) [reply tests/test_server_ops.py]: Fixed

**Review by elprans — APPROVED** [review]

**Pull request merged**